service-base/src/main/java/org/eclipse/hono/service/plan/TenantPlanValidator.java

```diff
@@ -0,0 +1,30 @@
+/*******************************************************************************
+ * Copyright (c) 2016, 2017 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.hono.service.plan;
+
+import io.vertx.core.Future;
+import org.eclipse.hono.util.TenantObject;
+
+/**
+ * Interface to check if further connections or messages are allowed based on the configured tenant limits.
+ */
+public abstract class TenantPlanValidator {
```

Comment:
why isn't this an interface?

Reply:
IMHO the overall purpose of this interface should not be tied to the notion of a *tenant*. In fact, the limits that need to be checked could as well be scoped to the whole system. I would therefore prefer a more neutral name. What about `ResourceLimitChecks`?

Reply:
@sophokles73 In the issue it is mentioned as per _tenant_ and so the scope. Hence I have named it so. In case of system level scope more neutral name like `ResourceLimitChecks` sounds good.